compiler/src/dotty/tools/dotc/ast/TreeInfo.scala

```diff
@@ -33,7 +33,11 @@ trait TreeInfo[T >: Untyped <: Type] { self: Trees.Instance[T] =>
   def defKind(tree: Tree): FlagSet = unsplice(tree) match {
     case EmptyTree | _: Import => NoInitsInterface
     case tree: TypeDef => if (tree.isClassDef) NoInits else NoInitsInterface
-    case tree: DefDef => if (tree.unforcedRhs == EmptyTree) NoInitsInterface else NoInits
+    case tree: DefDef =>
+      if (tree.unforcedRhs == EmptyTree)
+        (NoInitsInterface /: tree.vparamss.flatten)((fs, vparam) => fs & defKind(vparam))
```

Comment:
Yes, it's for default arguments, the corresponding commit message gives more information: https://github.com/lampepfl/dotty/pull/2062/commits/618fe9645b064b2a28be1505fe52cb3b94f7a25b

>  if a parameter has a default value the enclosing trait is still a NoInits
Ah indeed, I missed that.


> won't we then see the default method which gives us the right DefKind anyway?
No, because `defKind` is called on each element of the body of the class before they are desugared.

It would be much simpler if `defKind` was called after desugaring (and would also avoid other issues with default parameters of class constructors for example) but I don't know what that would involve.